hw/dv/verilator/cpp/scrambled_ecc32_mem_area.cc

```diff
@@ -3,20 +3,24 @@
 // SPDX-License-Identifier: Apache-2.0
 
 #include "scrambled_ecc32_mem_area.h"
-#include "scramble_model.h"
 
 #include <algorithm>
 #include <cassert>
 #include <iostream>
 #include <sstream>
 
+#include "scramble_model.h"
 #include "sv_scoped.h"
 
 // This is the maximum width of a nonce that's supported by the code in
 // prim_util_get_scramble_key_nonce.svh
 static const uint32_t kScrMaxNonceWidth = 320;
 static const uint32_t kScrMaxNonceWidthByte = (kScrMaxNonceWidth + 7) / 8;
 
+// Repeat the keystream of one single PRINCE instance if
+// set to true. Otherwise multiple PRINCE instances are used.
+static const bool kRepeatKeystream = true;
```

Comment:
Would you mind making this an argument to the `ScrambledEcc32MemArea` constructor? I don't think the result would be more complicated, and the "else" blocks below end up as dead code otherwise.

Reply:
No problem - moved into the constructor.